### PR TITLE
Add PackageCloud CLI

### PR DESCRIPTION
**Pull request by jcapona**
| Status  | Ticket |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/OS-1243) |

#### Main changes

Add CLI to manage PackageCloud repositories. It displays information about one/all packages and the available versions and cleaning up repositories from old versions.

#### Screenshots (feature, test output, profiling, dev tools etc)

N/A

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

**Review by angusjfw — COMMENTED** [review]
```
❯ package-cloud --help
Traceback (most recent call last):
  File "/usr/bin/package-cloud", line 33, in <module>
    sys.exit(load_entry_point('package-cloud-cli', 'console_scripts', 'package-cloud')())
  File "/usr/bin/package-cloud", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
  File "/usr/lib/python3.9/importlib/metadata.py", line 524, in distribution
    return Distribution.from_name(distribution_name)
  File "/usr/lib/python3.9/importlib/metadata.py", line 187, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: package-cloud-cli
```

File: scripts/package-cloud-cli/README.rst [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+     Deleting: 5.1.0-4
+     Deleting: 5.1.0-5
+     Deleting: 5.2.0-1
+  Kept versions: [5.2.1-1, 5.3.1-1]
+
+- Cleanup old versions of all packages
+
+.. code-block:: bash
+
+  $ package-cloud --all-packages --cleanup
```
Comment (angusjfw) [inline scripts/package-cloud-cli/README.rst]: in this example are no packages deleted because `--versions-to-keep` is not set? could be clearer

**Review by angusjfw — COMMENTED** [review]
this looks great. i wonder if we might want to do promotion through the cli too. could then create github actions to do it so all publishing is in github. i think it would be really useful if this cli could list packages that might need promotion.

maybe for now, when listing the available versions,  show the most stable repo that version was found in. eg
```
      Package: pt-os-net-mods (2 versions), latest: 1.0.1-2
      Versions: [1.0.1-1 (stable), 1.0.1-2 (experimental)]
or    Versions: { stable: [1.0.1-1], experimental: [1.0.1-2] }
````

File: scripts/package-cloud-cli/README.rst (hunk 1 above)
Comment (jcapona) [reply scripts/package-cloud-cli/README.rst]: I've combined the `--cleanup` and `--versions-to-keep` options into `--cleanup-and-keep` that will perform the cleanup leaving up to the provided number of versions in the repo.. the [README](https://github.com/pi-top/pi-topOS-Software/tree/OS-1243-add-package-cloud-cli/scripts/package-cloud-cli) should make it clear

**Comment by angusjfw** [comment]
> ```
> importlib.metadata.PackageNotFoundError: package-cloud-cli
> ```

i still get this error on my dev machine and pi-top, even when using `sudo -H`

**Review by angusjfw — APPROVED** [review]
